src/main/java/com/flightstats/hub/dao/aws/S3BatchContentDao.java

```diff
@@ -152,13 +153,15 @@ private Content getContent(ContentKey key, ZipInputStream zipStream, ZipEntry ne
         return builder.build();
     }
 
+    @SneakyThrows
     private ZipInputStream getZipInputStream(String channel, ContentPathKeys minutePath) {
         ActiveTraces.getLocal().add("S3BatchContentDao.getZipInputStream");
         long start = System.currentTimeMillis();
         try {
             GetObjectRequest request = new GetObjectRequest(bucketName, getS3BatchItemsKey(channel, minutePath));
-            S3Object object = s3Client.getObject(request);
-            return new ZipInputStream(new BufferedInputStream(object.getObjectContent()));
+            try (S3Object object = s3Client.getObject(request)) {
+                return new ZipInputStream(new BufferedInputStream(object.getObjectContent()));
```

Comment:
This is going to close the stream on exit.

https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/model/S3Object.java#L223

Reply:
True, tests are broken.